functions/auth0_cis_webhook_consumer/utils.py

```diff
@@ -36,69 +47,197 @@ def verify_token(authorization: str) -> bool:
+    """
+    prod_ldap_user_id_prefix = 'ad|Mozilla-LDAP|'
+    dev_ldap_user_id_prefix = 'ad|Mozilla-LDAP-Dev|'
+    logger.debug(
+        f"{user_id=} {CONFIG.management_api_discovery_document['issuer']=}")
+    if (user_id.startswith(prod_ldap_user_id_prefix)
+            and CONFIG.management_api_discovery_document['issuer'] == 'https://auth-dev.mozilla.auth0.com/'):
+        # Hack to translate LDAP user IDs fetched from production and added to
+        # Auth0 auth-dev
+        result = "{}{}".format(
+            dev_ldap_user_id_prefix,
+            user_id[len(prod_ldap_user_id_prefix):])
+        logger.debug('Hacking user_id from {} to {}'.format(user_id, result))
+        return result
+    else:
+        return user_id
+
+
+def process_auth0_user(user_id: str, operation: str) -> bool:
+    """Process the operation on the Auth0 user
+
+    Requires Auth0 Management API scopes
+    * update:users
+    * update:users_app_metadata
 
     :param user_id: The user's user ID
-    :param profile: The user's profile
+    :param operation: The operation to perform, "create", "update", "delete"
     :return:
     """
-    return True
+    if operation == "create":
+        # Would we ever want to trigger user creation in Auth0 because a CIS
+        # profile was created? I'd imagine not, as the Auth0 management API
+        # create only creates database and passwordless users
+        logger.debug(
+            "Ignoring request to create {} as we don't do Auth0 user "
+            "creation".format(user_id))
+        return True
+
+    if CONFIG.management_api_discovery_document is None:
+        return False
+    auth0_management_api_authorization = get_authorization(
+        CONFIG.management_api_discovery_document['token_endpoint'],
+        CONFIG.management_api)
+    if auth0_management_api_authorization is None:
+        return False
+    headers = {'authorization': auth0_management_api_authorization}
+    url = '{issuer}api/v2/users/{escaped_user_id}'.format(
+        issuer=CONFIG.management_api_discovery_document['issuer'],
+        escaped_user_id=urllib.parse.quote_plus(hack_user_id(user_id))
+    )
+
+    if operation == "delete":
+        # https://github.com/mozilla-iam/auth0-deploy/blob/8659ce4cef35ac22e459b35c09ffcc038b7f9bf8/rules/activate-new-users-in-CIS.js#L36
+        payload = {
+            'user_metadata': {
+                'existsInCIS': False
+            }
+        }
+        logger.info('Marking {} as existsInCIS False'.format(user_id))
+    elif operation == "update":
+        profile = get_user_profile(user_id)
+        if profile is None:
+            return False
+        access_groups = []
+        for publisher_name, data in profile.get(
+                'access_information', {}).items():
+            if data.get('values') is None:
+                continue
+            for value in data['values']:
+                if data['values'][value] is None:
```

Comment:
We use `""` as value for pmo managed mozilliansorg groups. Any particular reason for this check, or can we just omit it?

Reply:
@fiji-flo Ah! That's great to know.

Yes, the purpose of this check is to filter out content from the `hris` section of the profile

```json
{
    "access_information": {
        "hris": {
            "values": {
                "egencia_pos_country": "US",
                "employee_id": "1234",
                "managers_primary_work_email": "jdoe@mozilla.com",
                "primary_work_email": "jdoeboss@mozilla.com",
                "worker_type": "Employee"
            }
        }
    }
}
```

I didn't realize that there would be non `null` values in the `mozilliansorg` section. I don't see any in my and April's profile but maybe others have it? Anyhow, I'll change this check.

Reply:
@fiji-flo How does [this look](https://github.com/mozilla-iam/auth0-cis-webhook-consumer/pull/3/commits/ffc7205c3aba7bae85ad635220e4b950cc87b33b) as a solution?